### ISSUE TITLE
[FR] FR-012:  Make deepflow agent support for old linux kernel

### ISSUE THREAD

**Issue by eddietanshuo** [deepflowys/deepflow#2283]
### Search before asking

- [X] I had searched in the [issues](https://github.com/deepflowys/deepflow/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Our company has more than 1k host running on the linux 2.6. We met 2 problem when launched deepflow agent on these machine:

1. deepflow cgroup could not initialized:
```
[2023-01-28 17:51:19.232869 +08:00] WARN [src/[trident.rs:392](http://trident.rs:392/)] initialize cgroup controller failed, CpuControllerSetFailed("unable to write to a control group file caused by: Os { code: 22, kind: InvalidInput, message: \"Invalid argument\" }"), agent restart...
```
please check if deepflow-agent supports the kernel version without this commit: https://github.com/torvalds/linux/commit/74a1166dfe1135dcc168d35fa5261aa7e087011b

2. The agent did not calculate `available memory` correctly
![image](https://user-images.githubusercontent.com/103412360/215663040-e7a7bbcd-11c8-4ef5-ac98-68c871706096.png)
agent only works correctly after this commit: https://github.com/torvalds/linux/commit/34e431b0ae398fc54ea69ff85ec700722c9da773

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/deepflowys/deepflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by sharang** [comment]
In PR [#2287](https://github.com/deepflowys/deepflow/pull/2287), we added cgroup to adapt to kernel < 3, but due to the limitation of cgroup, we can only limit the resource consumption of the main thread.

Also for kernel 2.6, we removed the call to the free_memory_check function. Since the cgroup's memory consumption limit for the main thread is equivalent to the consumption limit for the entire process, the memory limit capability is still intact.

In the subsequent version of v6.2, we will consider scanning the thread list of deepflow-agent and refreshing it to the tasks file of cgroup to achieve the perfect CPU usage limit under kernel 2.6.
